simple_certmanager/validators.py

```diff
@@ -0,0 +1,41 @@
+from abc import ABC, abstractmethod
+
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+from OpenSSL import crypto
+
+
+class AbstractSSLValidator(ABC):
+    message = _("%(file)s does not contain a valid %(cert)s")
+    code = "invalid"
+    cert: str
+
+    @abstractmethod
+    def validate(self, file_type, file_content):
+        raise NotImplementedError
+
+    def __call__(self, value):
+        with value.open(mode="rb") as key_f:
+            try:
+                self.validate(crypto.FILETYPE_PEM, key_f.read())
+            except crypto.Error:
+                raise ValidationError(
+                    self.message,
+                    code=self.code,
+                    params={"file": value, "cert": self.cert},
+                )
+
+
+class PublicCertValidator(AbstractSSLValidator):
+    cert = "public_certificate"
+
+    def validate(self, file_type, file_content):
+        return crypto.load_certificate(file_type, file_content)
+
+
+class PrivateKeyValidator(AbstractSSLValidator):
+    cert = "private_key"
+
+    def validate(self, file_type, file_content):
+        return crypto.load_privatekey(file_type, file_content)
```

Comment:
```suggestion
class AbstractSSLValidator(ABC):
    message = _("Invalid file provided.")
    code = "invalid"

    @abstractmethod
    def validate(self, file_content: bytes):  # pragma: no cover
        raise NotImplementedError

    def __call__(self, value):
        with value.open(mode="rb") as infile:
            try:
                self.validate(infile.read())
            except crypto.Error:
                raise ValidationError(self.message, code=self.code)


class PublicCertValidator(AbstractSSLValidator):
    message = _("Invalid file provided, expected a certificate in PEM format.")

    @staticmethod
    def validate(file_content: bytes):
        return crypto.load_certificate(crypto.FILETYPE_PEM, file_content)


class PrivateKeyValidator(AbstractSSLValidator):
    message = _("Invalid file provided, expected a private key in PEM format.")

    @staticmethod
    def validate(file_content: bytes):
        return crypto.load_privatekey(crypto.FILETYPE_PEM, file_content)
```

* adds the appropriate type hints here and there
* makes the error message localize to the private key/public cert validation allowing for full contextual translation to NL
* the validate function itself can be simplified to a staticmethod
* moved the `crypto.FILETYPE_PEM` to make code easier to follow
* the use of `ABC` is a bit overkill here... we do not expect to expose this validator as public API